### PR TITLE
Use floodgate for ratelimiting

### PR DESCRIPTION
**Pull request by Lunarmagpie**
I tested it and im pretty confident it works

### CODE REVIEW

**Comment by Lunarmagpie** [comment]
I think this is blocked until hikari is updated to the newest version

**Comment by FuexFollets** [comment]
May I ask what the purpose is of using floodgate?

File: mcodingbot/plugins/highlights.py [hunk 1]
```diff
@@ -219,7 +215,7 @@ async def on_message(event: hikari.GuildMessageCreateEvent) -> None:
                 highlights[user_id].append(highlight)
 
     for user_id, hls in highlights.items():
-        if sent_message_cooldown.get_retry_after(
+        if sent_message_cooldown.trigger(
```
Comment (CircuitSacul) [inline mcodingbot/plugins/highlights.py:218]: This should be .retry_after

**Review by CircuitSacul — CHANGES_REQUESTED** [review]

**Comment by Lunarmagpie** [comment]
I merged this with upstream—CI passes so it should be reviewed

File: mcodingbot/plugins/highlights.py (hunk 1 above)
Comment (CircuitSacul) [reply mcodingbot/plugins/highlights.py:218]: This should be .retry_after, or `if not sent_message_cooldown.can_trigger`. It shouldn't actually be triggering anything.

File: mcodingbot/plugins/highlights.py [hunk 2]
```diff
@@ -205,7 +201,7 @@ async def on_message(event: hikari.GuildMessageCreateEvent) -> None:
 
     for highlight, users in highlights_cache.items():
         if highlight in event.content:
-            retry_after = trigger_cooldown.update_ratelimit(
+            retry_after = trigger_cooldown.retry_after(
```
Comment (CircuitSacul) [inline mcodingbot/plugins/highlights.py:204]: This should be .trigger

**Review by CircuitSacul — CHANGES_REQUESTED** [review]

**Review by CircuitSacul — APPROVED** [review]

**Review by trag1c — APPROVED** [review]

**Pull request merged**